test/parallel/test-cli-eval.js

```diff
@@ -229,3 +229,36 @@ child.exec(`${nodejs} --use-strict -p process.execArgv`,
     assert.strictEqual(err, null);
   }));
 });
+
+// ESModule eval tests
+
+
+// Assert that "42\n" is written to stdout on module eval.
+const execOptions = '--experimental-modules --input-type module';
+const esmWarning = 'ExperimentalWarning: ' +
+  'The ESM module loader is experimental.\n';
+child.exec(
```

Comment:
I'm a bit concerned that this test would start failing once the warning is removed. The most likely reaction is for somebody to remove the check for the warning and at that point this test no longer tests anything (effectively).

One way around it would be to use a module only feature or reference a CJS file feature:

```js
console.log(typeof import.meta); // will print "object" in module, fail in CJS
console.log(typeof require); // will print "function" in CJS, "undefined" in module
```

I don't think we'd want to check for the ESM warning itself since it's not part of the feature this test is targeting (eval of ESM).

Reply:
Got it! I'll update this PR with your comments